### PR TITLE
Case insensitive header key retrieval for asgi instrumentation

### LINKED ISSUE #307: Asgi request headers are not retrieved properly
Environment: 
Python: python3.8

fastapi==0.63.0
opentelemetry-api==0.16b1
opentelemetry-sdk==0.16b1
opentelemetry-instrumentation-fastapi==0.16b1
opentelemetry-exporter-google-cloud==0.16b1
opentelemetry-tools-google-cloud==0.16b1

When using `CloudTraceFormatPropagator` for [GCP](https://github.com/GoogleCloudPlatform/opentelemetry-operations-python), `X-Cloud-Trace-Context` header cannot be retrieved.

**Steps to reproduce**
Describe exactly how to reproduce the error. Include a code sample if applicable.

```
# server.py
import uvicorn
from fastapi import FastAPI, Request
from opentelemetry import trace
from opentelemetry.propagators import set_global_textmap
from opentelemetry.instrumentation.fastapi import FastAPIInstrumentor
from opentelemetry.sdk.trace import TracerProvider
from opentelemetry.tools import cloud_trace_propagator

set_global_textmap(cloud_trace_propagator.CloudTraceFormatPropagator())

app = FastAPI()

tracer = trace.get_tracer("test")
FastAPIInstrumentor.instrument_app(app)

@app.get("/trace")
async def test(r: Request):
    with tracer.start_as_current_span("test") as span:
        trace_id = span.get_span_context().trace_id
        print(f"{trace_id:32x}")  # should print trace ID from `X-Cloud-Trace-Context` header value

uvicorn.run(app)
```

```
# client.py
import requests

r = requests.Session()
r.headers.setdefault("X-Cloud-Trace-Context",
                        "f3ef5c2ede256aa77491057e600eca11/15104302039794794507;o=1")
r.get("http://localhost:8000/trace")
```

**What is the expected behavior?**
Printed value should be `f3ef5c2ede256aa77491057e600eca11` based from the header sent

**What is the actual behavior?**
A newly generated value everything `/trace` is called

**Additional context**
`X-Cloud-Trace-Context` header value is not retrieved properly in `CloudTraceFormatPropagator`

### PR DESCRIPTION
**Pull request by garnertan**
# Description

Made Asgi Instrumentor retrieve header keys in lower case.

Fixes #307 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

tox -e test-instrumentation-asgi

Added new test cases in test_getter.py

# Does This PR Require a Core Repo Change?

- [ ] Yes. - Link to PR: 
- [x] No.

# Checklist:

See [contributing.md](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/CONTRIBUTING.md) for styleguide, changelog guidelines, and more.

- [x] Followed the style guidelines of this project
- [x] Changelogs have been updated
- [x] Unit tests have been added
- [x] Documentation has been updated

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Remove `component` span attribute in instrumentations. 
   `opentelemetry-instrumentation-aiopg`, `opentelemetry-instrumentation-dbapi` Remove unused `database_type` parameter from `trace_integration` function.
   ([#301](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/301))
+- `opentelemetry-instrumentation-asgi` Return header values using case insensitive keys
```
Comment (toumorokoshi) [inline CHANGELOG.md:13]: make sure to add the PR link here too.

**Review by toumorokoshi — APPROVED** [review]
LGTM, thanks! Good catch.

**Review by codeboten — APPROVED** [review]
LGTM, just needs a quick update to the changelog as per @toumorokoshi's comment. Also please sign the CLA

**Comment by garnertan** [comment]
Thanks for reviewing. I added the pull request in the changelog.

**Pull request merged**